### ISSUE TITLE
Proposal: Include more relevant information when a media block upload fails

### ISSUE THREAD

**Issue by guarani** [WordPress/gutenberg#25590]
**Describe the proposal**

In the context of Gutenberg inside the WordPress iOS and Android apps, I'd like to propose adding more relevant information to the UI when an error occurs when uploading media (e.g. in an Image block).

Consider this scenario:
1. User opens the block editor, adds an Image block, tries to add an image and the server is unavailable (for whatever reason)
2. User sees the "Failed to insert media. Please tap for options." overlay.

Question: How can we show more relevant information to help diagnose the problem? How do we handle different kinds of errors and ensure we have control over what information is shown in the block (if it's text, are there localization issues? If it's just a code number such as "503", that won't be a problem AFAIK).

**To reproduce**
A server is unavailable error can be triggered by following these steps:
1. Log into a self-hosted Jurassic Ninja site
2. Go to the block editor, and add an Image block
3. The next two steps should be performed at the same time:
  a. Go to Jurassic Ninja (https://jurassic.ninja/) and create a new site
  b. Tap ADD IMAGE on the Image block and choose any image
4. Expect the image upload to fail due to the simultaneous site creation happening on Jurassic Ninja 
5. Notice that the error shown to the user is quite generic "Failed to insert media. Please tap for options."
6. (Optional) Still inside the app, if the user goes to the site's Media library, the image that failed to upload will be there (with a Retry button overlay)

The problems I see in the above is that the app doesn't show any specific information about the error that occurred. In the example above, the server returned a HTTP 503 error. Could we include this information in the block's error UI?

**Screen recording**

![RPReplay_Final1600895785 2020-09-23 17_18_34](https://user-images.githubusercontent.com/1898325/94070824-dfe38700-fdc0-11ea-8b4e-965d847fc807.gif)

**Editor version (please complete the following information):**
- WordPress version: [e.g: 5.3.2]
- Does the website have the Gutenberg plugin installed, or is it using the block editor that comes by default? [e.g: "gutenberg plugin", "default"]
- If the Gutenberg plugin is installed, which version is it? [e.g., 7.6]

**Smartphone (please complete the following information):**
 - Device: iPhone
 - OS: iPhone 14
 - WordPress App Version: 15.7

**Comment by geriux** [comment]
I was about to create a new issue related to this but I'll comment on this one instead. 

An issue was found while testing the Gallery block on Android, images would give upload errors but we are not showing exactly what happened:

<kbd><img src="https://user-images.githubusercontent.com/4885740/104168913-0fb1ab80-53ff-11eb-9fa5-2cd975ab6aaa.jpg" width=150 /></kbd>

### Steps to reproduce:

Using an Android device

- Log into a self-hosted Jurassic Ninja site
- Create a post and add a `Gallery` block
- Select a lot of images, around 25
- Expect images to start uploading and then some will start failing
- Tapping on a failed image, it will only show `We couldn't complete this action`

Checking the error log I found errors related to image resizing so in the end it was due to memory limits. We should at least show the PHP / Apache error so users are able to troubleshoot.

**Comment by guarani** [comment]
I agree @geriux, even if it's an error code which the user may not understand, it's helpful when troubleshooting with users.

**Comment by SiobhyB** [comment]
Noting that I'm going to be taking a look into how we can improve the error messages for failed media uploads. :) Here's an attempt to gather a list of related open issues in one place for reference:

## Gutenberg Mobile repository

* https://github.com/wordpress-mobile/gutenberg-mobile/issues/2448

## Android repository

* https://github.com/wordpress-mobile/WordPress-Android/issues/12325
* https://github.com/wordpress-mobile/WordPress-Android/issues/5109
* https://github.com/wordpress-mobile/WordPress-Android/issues/5158
* https://github.com/wordpress-mobile/WordPress-Android/issues/13815

## iOS repository

* https://github.com/wordpress-mobile/WordPress-iOS/issues/14400
* https://github.com/wordpress-mobile/WordPress-iOS/issues/14401
* https://github.com/wordpress-mobile/WordPress-iOS/issues/17468
* https://github.com/wordpress-mobile/WordPress-iOS/issues/14397
* https://github.com/wordpress-mobile/WordPress-iOS/issues/12793

## Prior Work

* https://github.com/wordpress-mobile/WordPress-Android/issues/5167
* https://github.com/wordpress-mobile/WordPress-Android/issues/7103
* https://github.com/wordpress-mobile/WordPress-Android/pull/12672

**Comment by tyxla** [comment]
This is no longer relevant after the React Native implementation was removed in #78747. Closing.
